### PR TITLE
feat(geo_pose_projector): use geo_pose_projector in eagleye

### PR DESCRIPTION
**Pull request by kminoda**
## Description

Since current Eagleye implementation does not align with the `/map/map_projector_info`, I would like to propose the following modification
- Adding a new package `tier4_geo_pose_projector` that, given a georeferenced pose and the projector info, converts and publishes the pose
- Use this in Eagleye instead of `geo_pose_converter` from Eagleye repository

Note that the new package will be added to autoware.universe instead of Eagleye repository, since it is dependent on Autoware's component interface and may be used to other applications as well.

## Tests performed

- Confirmed that, using logging_simulator, eagleye works with sample map from shinjuku AWSIM data
- Confirmed that, using logging_simulator and the following map_projector_info.yaml, eagleye works with sample map from shinjuku AWSIM data

```YAML
projector_type: TransverseMercator
vertical_datum: WGS84
map_origin:
  latitude: 35.61836750154
  longitude: 139.78066608243
  altitude: 2.0
```

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by YamatoAndo** [comment]
@kminoda I think that this PR would resolve [this issue](https://github.com/autowarefoundation/autoware.universe/issues/2915)? Is that correct?

**Comment by KYabuuchi** [comment]
> ## Tests performed
> * Confirmed that, using logging_simulator, eagleye works with sample map from shinjuku AWSIM data
> * Confirmed that, using logging_simulator and the following map_projector_info.yaml, eagleye works with sample map from shinjuku AWSIM data

I have confirmed them in my environment. They works properly.

File: localization/geo_pose_projector/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+| ------------------------- | ---------------------------------------------------- | ------------------- |
+| `input_geo_pose`          | `geographic_msgs::msg::GeoPoseWithCovarianceStamped` | geo-referenced pose |
+| `/map/map_projector_info` | `tier4_map_msgs::msg::MapProjectedObjectInfo`        | map projector info  |
+
+## Published Topics
+
+| Name          | Type                                            | Description       |
+| ------------- | ----------------------------------------------- | ----------------- |
+| `output_pose` | `geometry_msgs::msg::PoseWithCovarianceStamped` | pose in map frame |
+
```
Comment (YamatoAndo) [inline localization/geo_pose_projector/README.md:19]: please add `tf` topic

Comment (kminoda) [reply localization/geo_pose_projector/README.md:20]: Thanks, done: https://github.com/autowarefoundation/autoware.universe/pull/5513/commits/46e7ede43b47b3cae31664b9be81ffcd298e956e

**Review by YamatoAndo — APPROVED** [review]

**Pull request merged**